src/app/modules/fs/controllers/fsResult.js

```diff
@@ -162,38 +166,49 @@ fsModule.controller('FsResultCtrl', [
       $scope.timerScope.percent = 0
       $scope.seconds = '-'
       $scope.numTry = $scope.numTryLimit
+      $scope.showCancelRequest = false
+      $scope.consentCheck = '-'
     }
 
     $scope.tryAgainNow = function (e) {
       $scope.checkConsent()
     }
 
     $scope.checkConsent = function () {
-    // reset the seconds countdown and progress bar
+      // reset the seconds countdown and progress bar
       $scope.timerScope.percent = 0
       $scope.seconds = '-'
+      $scope.stateReason = FsInfoService.t('checkingNow')
 
-    // send the consent API request
+      // send the consent API request
       FsBankService.sendConsentRequest(fs).then(function (data) {
-      // start the timer again
+        // start the timer again
         fs.consentResponse = data
+        $scope.stateReason = FsInfoService.t('consentPendingReason')
         var consentStatus = FsBankService.getConsentStatus(fs)
         FsService.track('consent', consentStatus, 'financial-status')
         if (data.data.consent === 'SUCCESS') {
           $scope.cancelTimer()
+
+          $scope.stateReason = FsInfoService.t('consentGivenReason')
+          $scope.consentCheck = FsInfoService.t('checkingBalance')
+
           $timeout(function () {
             $scope.checkBalance()
-          }, 60000)
+          }, 500)
         } else if (data.data.consent === 'FAILURE' || data.data.consent === 'INVALID') {
```

Comment:
I know this hasn't changed, but I think we should differentiate between a genuine failure and a failure due to invalid data.  Can we please check this with someone from product?